### DISCUSSION TITLE
Suggestion: SourceGenerator EmbeddedResource API

### DISCUSSION THREAD

**Discussion by MarkPflug** [dotnet/roslyn#49923] [Ideas] [answered]
I have a project that I maintain that is implements .resx-like functionality via json files (as opposed to xml). The project uses MSBuild tasks to generate C# resource accessor classes, and also to generated embedded resources. I'd love to migrate this project to use source generators instead. Unfortunately, it appears that the source generator API doesn't currently support adding embedded resources to the compilation.

Is that a feature that could be considered for a future release? It might be as simple as: GeneratorExecutionContext.AddEmbeddedResource(byte[], string logicalName);

But, of course, I don't know the inner workings of the compiler, so there's quite likely more complexity than I'm imagining.

**Comment by jasonmalinowski** [comment, marked as answer]
@chsienki Do we already have a issue tracking this?

**Reply by chsienki** [reply]
No, but interestingly enough this came up yesterday from the runtime team too. I'll open a tracking issue for it

**Reply by chsienki** [reply]
Opened https://github.com/dotnet/roslyn/issues/49935 to track this.

**Reply by MarkPflug** [reply]
@jasonmalinowski @chsienki Question meta-question: There was an issue I opened a while back (https://github.com/dotnet/roslyn/issues/49260) that I closed, thinking it was resolved, then reopened when I realized it wasn't. Will that end up being abandoned since it was closed? I don't know how these workflows work, so I'm just curious if that issue is going to sit unattended forever. Not trying to pester you, I know you have higher priority work than helping me with my issues.

Thanks.

**Reply by jasonmalinowski** [reply]
Friendly pestering is always OK! Unfortunately sometimes things get buried with all the usual chaos so a friendly ping often helps to elevate stuff.

(as far as the issue itself, that's @chsienki's area of expertise so I'm useless there. :smile:)

**Reply by lennybacon** [reply]
Friendly pestering 2025 edition :-) Any updates?

**Reply by CyrusNajmabadi** [reply]
No updates on this. When there are updates we will report as much on these issues. Thanks.
